instructions/docs/add-the-encryption-sdk.md

```diff
@@ -0,0 +1,230 @@
+    # ADD-ESDK-START
+    encrypted_data, header = aws_encryption_sdk.encrypt(
+        source=data,
+        key_provider=self.master_key_provider,
+    )
+```
+
+#### What Just Happened
+
+You just started encrypting data client-side with the AWS Encryption SDK and KMS!
+
+Now, before storing your data in the Document Bucket, you use the AWS Encryption SDK to:
+
+1. Request a new data key using your keyring or Master Key Provider
+1. Encrypt that data for you
+1. Return the encrypted data in the AWS Encryption SDK message format
+1. Extract the ciphertext to pass to the AWS S3 SDK to store in S3
+
+### Step 3: Add Decryption to `retrieve`
+
+Now that you are encrypting data before storing it, you need to decrypt it before returning it to your caller. At least for it to be useful, anyway.
+
+```NodeJS
+TODO
+```
+
+#### What Just Happened
+```Python
+# Edit src/document_bucket/api.py
+    def retrieve(
+            self,
+            pointer_key: str,
+            expected_context_keys: Set[str] = set(),
+            expected_context: Dict[str, str] = {},
+    ) -> DocumentBundle:
+        item = self._get_pointer_item(PointerQuery.from_key(pointer_key))
+        # ADD-ESDK-START
+        encrypted_data = self._get_object(item)
+        plaintest, header = aws_encryption_sdk.decrypt(
+            source=encrypted_data, key_provider=self.master_key_provider
+        )
+        return DocumentBundle.from_data_and_context(
+            plaintext, item.context
+        )
+```
+
+
+You just started decrypting data client-side as well!
+
+The data returned from S3 for `retrieve` is now encrypted. Before returning that data to the user, you added a call to the AWS Encryption SDK to decrypt the data. Under the hood, the Encryption SDK:
+
+1. Read the AWS Encryption SDK formatted encrypted message
+1. Called KMS to request to decrypt your message's encrypted data key using the Faythe CMK
+1. Used the decrypted data key to decrypt the message
+1. Returned the message plaintext and Encryption SDK headers to you
+
+
+### Step 4: Plumb In Your Config
+
+Now that you have your dependencies declared and your code updated to encrypt and decrypt data, all you need to do is pass through the configuration to the AWS Encryption SDK to start using your KMS CMKs to protect your data.
+
+```NodeJS
+
+TODO
+
+```
+
+```Python
+
+# Edit src/document_bucket/__init__.py
+
+...
+
+# ADD-ESDK-START
+# Pull configuration of KMS resources
+faythe_cmk = state["FaytheCMK"]
+# And the Master Key Provider configuring how to use KMS
+cmk = [faythe_cmk]
+mkp = aws_encryption_sdk.KMSMasterKeyProvider(key_ids=[cmk])
+
+...
+
+operations = DocumentBucketOperations(bucket, table, mkp)
+```
+
+#### What Just Happened
+
+In Getting Started, you launched CloudFormation stacks for CMKs. One of these CMKs was nicknamed Faythe. As part of launching these templates, the CMK's ARN was written to a configuration file on disk, the `state` variable that is loaded and parsed.
```

Comment:
if this is the first time we've used "ARN" we should consider spelling out Amazon Resource Name ("ARN")

Reply:
Good catch, fixed.